### PR TITLE
MILAB-6225: Add federative envelope handling to Ls API

### PR DESCRIPTION
**Pull request by vgpopov**
- Extend Ls API to carry additionalInfo envelope for federative storages
- Switch list results to a file-stats variant and propagate envelopes
- Pass envelope through to index handles when present
- Update REST/GRPC schemas to include additionalInfo
- Add tests to verify envelope threading and absence when empty

<details><summary><h3>Greptile Summary</h3></summary>

This PR extends the Ls API to carry a federative identity envelope (`additionalInfo`, a `map<string, string>`) through the full stack — proto definition, gRPC and REST codegen, TypeScript client, and index handle serialisation — so that downstream consumers can verify signed identity envelopes from federative storages.

- **Proto / schema layer**: `additional_info = 8` added to `LsAPI_ListItem` in `.proto`, OpenAPI YAML, and both generated TypeScript stubs (gRPC and REST).
- **Handle serialisation**: `createIndexImportHandle` now accepts an optional `additionalInfo` map and only embeds it when non-empty, keeping existing URLs byte-identical.
- **Driver rename + threading**: `listRemoteFilesWithAdditionalInfo` is renamed to `listRemoteFilesWithFileStats`; both `listFiles` and `listRemoteFilesWithFileStats` now forward `e.additionalInfo` to the handle builder.
</details>

<details open><summary><h3>Confidence Score: 3/5</h3></summary>

The data flow and serialisation are correct; the hand-written protobuf map decoder in protocol.ts will throw a hard error on any unrecognised field number, which is unsafe for forward compatibility.

The hand-written `binaryReadMap8` in the generated gRPC file unconditionally throws on unknown field numbers rather than skipping them. Every other field in the same generated file respects `options.readUnknownField`. Under normal proto version alignment this path never triggers, but it is a latent runtime breakage that needs fixing before this decoder is relied on in production for federative storages.

lib/node/pl-drivers/src/proto-grpc/github.com/milaboratory/pl/controllers/shared/grpc/lsapi/protocol.ts — the `binaryReadMap8` default case needs attention.
</details>

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| lib/node/pl-drivers/src/proto-grpc/github.com/milaboratory/pl/controllers/shared/grpc/lsapi/protocol.ts | Adds `additionalInfo` map field (field 8) to `LsAPI_ListItem` with a hand-written `binaryReadMap8` decoder. The decoder throws on any unrecognised field number instead of skipping it per proto3 spec, which could break parsing under version skew. |
| lib/node/pl-drivers/src/drivers/helpers/ls_remote_import_handle.ts | Adds optional `additionalInfo` parameter to `createIndexImportHandle`; only embeds it when non-empty, preserving byte-identical URLs for non-federative storages. |
| lib/node/pl-drivers/src/drivers/ls.ts | Renames `listRemoteFilesWithAdditionalInfo` to `listRemoteFilesWithFileStats` and threads `e.additionalInfo` into both list methods; two TODO-4 comments remain in production call sites. |
| lib/node/pl-drivers/src/drivers/types.ts | Adds optional `additionalInfo: z.record(z.string(), z.string())` to `ImportFileHandleIndexData` Zod schema; backwards compatible. |
| lib/node/pl-drivers/src/drivers/ls.test.ts | Extends existing test file with `LsDriver additionalInfo threading` suite covering envelope propagation in both list methods. |

</details>

</details>

<details><summary><h3>Sequence Diagram</h3></summary>

```mermaid
sequenceDiagram
    participant Backend as gRPC/REST Backend
    participant Client as ClientLs (ls_api.ts)
    participant Driver as LsDriver (ls.ts)
    participant Handle as createIndexImportHandle
    participant Consumer as Downstream Consumer

    Backend->>Client: "LsAPI_ListItem { additionalInfo, fullName, ... }"
    Client->>Driver: items with additionalInfo map
    Driver->>Handle: createIndexImportHandle(storageId, fullName, additionalInfo)
    Note over Handle: non-empty? embed in JSON, else omit
    Handle-->>Driver: index://index/encoded-JSON
    Driver-->>Consumer: "LsEntry { handle, type, name, fullPath }"
    Consumer->>Consumer: "parseIndexHandle -> { storageId, path, additionalInfo? }"
```
</details>

<a href="https://app.greptile.com/ide/claude-code?prompt=Fix%20the%20following%202%20code%20review%20issues.%20Work%20through%20them%20one%20at%20a%20time%2C%20proposing%20concise%20fixes.%0A%0A---%0A%0A%23%23%23%20Issue%201%20of%202%0Alib%2Fnode%2Fpl-drivers%2Fsrc%2Fproto-grpc%2Fgithub.com%2Fmilaboratory%2Fpl%2Fcontrollers%2Fshared%2Fgrpc%2Flsapi%2Fprotocol.ts%3A242-257%0A**Unconditional%20throw%20on%20unknown%20fields%20breaks%20proto3%20forward%20compatibility**%0A%0AThe%20%60default%60%20case%20throws%20unconditionally%20regardless%20of%20the%20%60options.readUnknownField%60%20configuration%2C%20while%20the%20outer%20%60internalBinaryRead%60%20properly%20gates%20on%20%60options.readUnknownField%60.%20Any%20wire%20format%20that%20carries%20an%20unexpected%20field%20number%20in%20the%20%60additional_info%60%20map%20entry%20%E2%80%94%20due%20to%20proto%20version%20skew%2C%20future%20extensions%2C%20or%20a%20misbehaving%20upstream%20%E2%80%94%20will%20throw%20a%20hard%20error%20instead%20of%20skipping%20the%20field%20as%20proto3%20requires.%20The%20%60options%60%20argument%20is%20accepted%20but%20ignored%20in%20this%20path.%20The%20standard%20fix%20is%20to%20call%20%60reader.skip%28wireType%29%60%20%28with%20optional%20%60UnknownFieldHandler.onRead%60%29%20rather%20than%20throwing.%0A%0A%23%23%23%20Issue%202%20of%202%0Alib%2Fnode%2Fpl-drivers%2Fsrc%2Fdrivers%2Fls.ts%3A220-221%0AThese%20%60TODO-4%60%20references%20are%20left%20in%20production%20code%20paths%20and%20will%20appear%20in%20every%20%60listFiles%60%20and%20%60listRemoteFilesWithFileStats%60%20call.%20If%20this%20work%20item%20has%20a%20separate%20tracking%20ticket%2C%20the%20comment%20should%20reference%20that%20ticket%20explicitly%3B%20if%20it%20was%20already%20resolved%20in%20this%20PR%2C%20the%20TODO%20can%20be%20removed.%0A%0A%60%60%60suggestion%0A%20%20%20%20%20%20%20%20%20%20handle%3A%20createIndexImportHandle%28storageData.storageId%2C%20e.fullName%2C%20e.additionalInfo%29%2C%0A%60%60%60%0A%0A&repo=milaboratory%2Fplatforma&pr=1647&platform=github"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInClaudeDark.svg?v=3"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInClaude.svg?v=3">Fix All in Claude Code</picture></a>

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
Fix the following 2 code review issues. Work through them one at a time, proposing concise fixes.

---

### Issue 1 of 2
lib/node/pl-drivers/src/proto-grpc/github.com/milaboratory/pl/controllers/shared/grpc/lsapi/protocol.ts:242-257
**Unconditional throw on unknown fields breaks proto3 forward compatibility**

The `default` case throws unconditionally regardless of the `options.readUnknownField` configuration, while the outer `internalBinaryRead` properly gates on `options.readUnknownField`. Any wire format that carries an unexpected field number in the `additional_info` map entry — due to proto version skew, future extensions, or a misbehaving upstream — will throw a hard error instead of skipping the field as proto3 requires. The `options` argument is accepted but ignored in this path. The standard fix is to call `reader.skip(wireType)` (with optional `UnknownFieldHandler.onRead`) rather than throwing.

### Issue 2 of 2
lib/node/pl-drivers/src/drivers/ls.ts:220-221
These `TODO-4` references are left in production code paths and will appear in every `listFiles` and `listRemoteFilesWithFileStats` call. If this work item has a separate tracking ticket, the comment should reference that ticket explicitly; if it was already resolved in this PR, the TODO can be removed.

```suggestion
          handle: createIndexImportHandle(storageData.storageId, e.fullName, e.additionalInfo),
```

`````

</details>

<sub>Reviews (1): Last reviewed commit: ["\[MILAB-6225\]: Add federative envelope ha..."](https://github.com/milaboratory/platforma/commit/504a380558567133e98a495f63ecd7cd59e091be) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=32837115)</sub>

> Greptile also left **2 inline comments** on this PR.

### CODE REVIEW

File: lib/node/pl-drivers/src/proto-grpc/github.com/milaboratory/pl/controllers/shared/grpc/lsapi/protocol.ts [hunk 1]
```diff
@@ -224,6 +239,22 @@ class LsAPI_ListItem$Type extends MessageType<LsAPI_ListItem> {
...
+                    key = reader.string();
+                    break;
+                case 2:
+                    val = reader.string();
+                    break;
+                default: throw new globalThis.Error("unknown map entry field for MiLaboratories.Controller.Shared.LsAPI.ListItem.additional_info");
+            }
+        }
+        map[key ?? ""] = val ?? "";
+    }
```
[the remark this answers is not in the document]
Comment (vgpopov) [reply lib/node/pl-drivers/src/proto-grpc/github.com/milaboratory/pl/controllers/shared/grpc/lsapi/protocol.ts:257]: generated file

**Review by DenKoren — APPROVED** [review]
